### PR TITLE
Create TAPESTRY_HOST and TAPESTRY_PORT env vars and move logs to a file

### PR DESCRIPTION
**Pull request by romulodl**
Tapestry will not listen in the heroku port anymore. Nginx will and it will proxy pass to tapestry running on port 3000.

Added a LOG_PATH where the `access.log` and `error.log` will live in the server, this 2 files will be redirected to `stdout` which is where `heroku logs` reads from.

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ class Server {
     // Create Hapi server instance
     const server = Hapi.server({
       host: process.env.HOST || '0.0.0.0',
-      port: process.env.PORT || 3000,
+      port: 3000,
```
Comment (thomasdigby) [inline src/server/index.js]: Why are you removing `process.env.PORT`?

Comment (romulodl) [reply src/server/index.js]: updated the description just now

**Comment by thomasdigby** [comment]
`No description provided.`

Talk to me, what is this thing, what does it do.

File: src/server/index.js (hunk 1 above)
Comment (romulodl) [reply src/server/index.js]: if we have to use a port number we should use a different name than PORT. As nginx will use the heroku PORT

**Comment by thomasdigby** [comment]
I'd like to retain the `PORT` env anyway, there's no need to remove it if it's not set. Will this affect the multiple instances of Tapestry run on production?

File: src/config/hot.js [hunk 2]
```diff
@@ -9,13 +9,13 @@ if (module.hot) {
     currentApp = createNewServerProxy()
     await registerPluginProxy(currentApp)
     await currentApp.start()
-    notify('🔁  HMR Reloading `./hot-server`...')
+    log.debug('🔁  HMR Reloading `./hot-server`...')
```
Comment (thomasdigby) [inline src/config/hot.js]: Let's update these to `log.info`

File: bin/start.js [hunk 3]
```diff
@@ -19,6 +19,8 @@ if (process.env.ENABLE_CONCURRENCY) {
   const WORKERS =
     process.env.WEB_CONCURRENCY || require('os').cpus().length || 1
   throng(WORKERS, () => server())
+  log.debug('Started ' + process.env.WEB_CONCURRENCY + ' processes')
 } else {
   server()
+  log.debug('Started server')
```
Comment (thomasdigby) [inline bin/start.js]: Same with these, feels very `log.info`

File: src/config/production-server.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ export default async function() {
     await registerPlugins({ config: appConfig, server: currentApp })
     // Start server
     await currentApp.start()
-    notify(`Server started at: ${currentApp.info.uri}\n`)
+    log.debug(`Server started at: ${currentApp.info.uri}\n`)
```
Comment (thomasdigby) [inline src/config/production-server.js]: `log.info`

File: src/server/utilities/logger.js [hunk 5]
```diff
@@ -9,24 +9,15 @@ const log = createLogger({
...
-    process.env.PORT !== process.env.TAPESTRY_PORT
+    process.env.NODE_ENV !== 'development'
     ? [
       new transports.File({ filename: process.env.LOG_PATH + '/error.log', level: 'error' }),
       new transports.File({ filename: process.env.LOG_PATH + '/access.log' })
     ]
     : [
-      new transports.File({ filename: '/dev/stderr', level: 'error' }),
-      new transports.File({ filename: '/dev/stdout', level: 'error' }),
+      new transports.Console({ format: format.simple() })
```
Comment (thomasdigby) [inline src/server/utilities/logger.js]: Nice. Worth flipping this around to `process.env.NODE_ENV === 'production'`?

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "tapestry-lite",
-  "version": "4.4.0",
+  "version": "4.5.0",
```
Comment (thomasdigby) [inline package.json]: Let's 5.0 it 🙆‍♂

File: src/server/utilities/logger.js [hunk 7]
```diff
@@ -1,25 +1,23 @@
...
-  transports: [
-    new winston.transports.Console({
-      timestamp: tsFormat,
-      colorize: true,
-      prettyPrint: true
-    })
-  ]
+  format: combine(timestamp(), logFormat),
+  transports:
+    typeof process.env.TAPESTRY_LOGS_FOLDER !== 'undefined'
```
Comment (thomasdigby) [inline src/server/utilities/logger.js]: Sorry... can we update this to `LOG_DIRECTORY` or `LOG_DIR` to be more consistent with `LOG_LEVEL`?

**Pull request merged**